api/v1alpha1/dspipeline_types.go

```diff
@@ -381,9 +381,23 @@ type SecretKeyValue struct {
 }
 
 type DSPAStatus struct {
+	// +kubebuilder:validation:Optional
+	Components ComponentStatus    `json:"components,omitempty"`
```

Comment:
I haven't seen this pattern before, but it seems to work well for this purpose. Was there an existing project that you used for inspiration?

Reply:
I used this as reference: https://gist.github.com/spolti/00afcb7252924bb21e61c60f63c521c6#file-gistfile1-txt-L22

Gage asked to replace `address` to `components`, and I grouped all endpoints in this place.